test/statsd/statsd_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build !windows
+
+package statsd
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric/dimension"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+)
+
+const namespace = "StatsD"
+
+type MetricBenchmarkTestSuite struct {
```

Comment:
Why do we need to copy paste all of this? Under the hood, the testing is just running `go test` on the specified directory.

Reply:
let me do some clean ups

Reply:
If we need to make some interfaces public so we can import them, I would prefer that rather than copying entire structs over to a different directory to serve the same purpose. I think that we need to do a refactor for the code structure in these tests, but don't know how or when we would do so. I'll make a GitHub issue for it since that's just how I like to work. We can mirror something internally for tracking if we want but I don't expect that we'd carve out time for it anytime soon.

Reply:
I carved some codes and pull out duplicates to a util eg to `common/metrics`